Detectors/CTP/workflow/src/RawDecoderSpec.cxx

```diff
@@ -257,29 +257,45 @@ int RawDecoderSpec::addCTPDigit(uint32_t linkCRU, uint32_t triggerOrbit, gbtword
   pld >>= 12;
   CTPDigit digit;
   const gbtword80_t bcidmask = 0xfff;
-  uint32_t bcid = (diglet & bcidmask).to_ulong();
+  int32_t bcid = (diglet & bcidmask).to_ulong();
   // LOG(info) << bcid << "    pld:" << pld;
   o2::InteractionRecord ir;
-  ir.orbit = triggerOrbit;
-  ir.bc = bcid;
-  digit.intRecord = ir;
+  int32_t BCShiftCorrection = o2::ctp::TriggerOffsetsParam::Instance().customOffset[o2::detectors::DetID::CTP];
   if (linkCRU == o2::ctp::GBTLinkIDIntRec) {
     LOG(debug) << "InputMaskCount:" << digits[ir].CTPInputMask.count();
+    if (bcid > BCShiftCorrection) {
```

Comment:
what if BCShiftCorrection is negative and `bcid - BCShiftCorrection` exceeds 3564?
Use simply 
```
if (it.toLong()>BCShiftCorrection) {
  ir -= BCShiftCorrection;
} else {
  ir = {0,0}; // normally should not happen
}
```

Reply:
I see your point although not sure what do you suggest.
I implemented now general formula.

Reply:
I mean the InteractionRecord class has += and -= operators (the argument is BC unit), one should just watch that when adding negative number the result (as ulong abs BC) remain positive since otherwise it will wrap to large value

Reply:
ok, I see now. So I used +=. 
Also I added correction for negative orbit/bcid to:
 InteractionRecord& operator+=(int64_t dbc)
https://github.com/lietava/AliceO2/blob/58055cd4486be4cef7125b0330582ef0644a51b8/DataFormats/common/include/CommonDataFormat/InteractionRecord.h#L200
I know in practice it may never happen. So it can be removed if there are other reasons not to have it.